### PR TITLE
Housekeeping!

### PR DESCRIPTION
**Pull request by JamesMGreene**
This is the separate housekeeping changes that were previously a part of PR #349.

### CODE REVIEW

File: lib/datastore.js [hunk 1]
```diff
@@ -535,9 +540,9 @@ Datastore.prototype._update = function (query, updateQuery, options, cb) {
   }
   , function () {   // Perform the update
     var modifiedDoc
-	  , candidates = self.getCandidates(query)
-	  , modifications = []
-	  ;
+      , candidates = self.getCandidates(query)
```
Comment (JamesMGreene) [inline lib/datastore.js]: Some random tabs that should've been spaces.

File: lib/datastore.js [hunk 2]
```diff
@@ -109,11 +109,16 @@ Datastore.prototype.resetIndexes = function (newData) {
...
+  var err
+    , callback = cb || function () {};
 
   options = options || {};
 
-  if (!options.fieldName) { return callback({ missingFieldName: true }); }
+  if (!options.fieldName) {
+    err = new Error("Cannot create an index without a fieldName");
+    err.missingFieldName = true;
+    return callback(err);
```
Comment (JamesMGreene) [inline lib/datastore.js]: Using proper `Error` instances, a la previous PR #341.

File: lib/indexes.js [hunk 3]
```diff
@@ -230,28 +230,20 @@ Index.prototype.revertUpdate = function (oldDoc, newDoc) {
...
-// Append all elements in toAppend to array
-function append (array, toAppend) {
-  var i;
-
-  for (i = 0; i < toAppend.length; i += 1) {
-    array.push(toAppend[i]);
-  }
-}
-
-
```
Comment (JamesMGreene) [inline lib/indexes.js]: Removed some plain ole dead code.

File: lib/storage.js [hunk 4]
```diff
@@ -59,11 +59,13 @@ storage.flushToStorage = function (options, callback) {
...
+        var e = null;
         if (errFS || errC) {
-          return callback({ errorOnFsync: errFS, errorOnClose: errC });
-        } else {
-          return callback(null);
+          e = new Error('Failed to flush to storage');
+          e.errorOnFsync = errFS;
+          e.errorOnClose = errC;
         }
+        return callback(e);
```
Comment (JamesMGreene) [inline lib/storage.js]: Using proper `Error` instances, a la previous PR #341.

File: test/db.test.js [hunk 5]
```diff
@@ -89,7 +89,7 @@ describe('Database', function () {
       db = new Datastore({ filename: autoDb, autoload: true, onload: onload })
 
       db.find({}, function (err, docs) {
-        done("Find should not be executed since autoload failed");
+        done(new Error("Find should not be executed since autoload failed"));
```
Comment (JamesMGreene) [inline test/db.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/db.test.js [hunk 6]
```diff
@@ -390,7 +390,7 @@ describe('Database', function () {
      *
      * Note: maybe using an in-memory only NeDB would give us an easier solution
      */
-    it('If the callback throws an uncaught execption, dont catch it inside findOne, this is userspace concern', function (done) {
+    it('If the callback throws an uncaught exception, do not catch it inside findOne, this is userspace concern', function (done) {
```
Comment (JamesMGreene) [inline test/db.test.js]: Fixed a spelling error.

File: test/db.test.js [hunk 7]
```diff
@@ -399,11 +399,13 @@ describe('Database', function () {
...
 
       process.on('uncaughtException', function MINE (ex) {
+        process.removeAllListeners('uncaughtException');
+
         for (i = 0; i < currentUncaughtExceptionHandlers.length; i += 1) {
           process.on('uncaughtException', currentUncaughtExceptionHandlers[i]);
         }
 
-        ex.should.equal('SOME EXCEPTION');
+        ex.message.should.equal('SOME EXCEPTION');
```
Comment (JamesMGreene) [inline test/db.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/db.test.js [hunk 8]
```diff
@@ -411,9 +413,9 @@ describe('Database', function () {
         d.findOne({ a : 5}, function (err, doc) {
           if (tryCount === 0) {
             tryCount += 1;
-            throw 'SOME EXCEPTION';
+            throw new Error('SOME EXCEPTION');
```
Comment (JamesMGreene) [inline test/db.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/db.test.js [hunk 9]
```diff
@@ -411,9 +413,9 @@ describe('Database', function () {
...
           } else {
-            done('Callback was called twice');
+            done(new Error('Callback was called twice'));
```
Comment (JamesMGreene) [inline test/db.test.js]: Using proper `Error` instances, a la previous PR #341.

**Comment by lukasbestle** [comment]
You don't have to comment every change. :D
If you need comments, use different commits I guess.

File: test/db.test.js [hunk 10]
```diff
@@ -399,11 +399,13 @@ describe('Database', function () {
       process.removeAllListeners('uncaughtException');
 
       process.on('uncaughtException', function MINE (ex) {
+        process.removeAllListeners('uncaughtException');
```
Comment (JamesMGreene) [inline test/db.test.js]: This was causing me a lot of grief while trying to debug actual test errors as they were all getting funneled into this `MINE` listener.

File: test/executor.test.js [hunk 11]
```diff
@@ -15,25 +15,26 @@ var should = require('chai').should()
...
 
   process.on('uncaughtException', function (err) {
     // Do nothing with the error which is only there to test we stay on track
   });
 
-  d.find({}, function (err) {    
+  d.find({}, function (err) {
     process.nextTick(function () {
       d.insert({ bar: 1 }, function (err) {
+        process.removeAllListeners('uncaughtException');
```
Comment (JamesMGreene) [inline test/executor.test.js]: Keeping the listeners tidy.

File: test/executor.test.js [hunk 12]
```diff
@@ -15,25 +15,26 @@ var should = require('chai').should()
...
         }
-        
+
         done();
       });
     });
-    
-    throw 'Some error';
+
+    throw new Error('Some error');
```
Comment (JamesMGreene) [inline test/executor.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/executor.test.js [hunk 13]
```diff
@@ -51,29 +52,34 @@ function testRightOrder (d, done) {
...
           process.nextTick(function () {
             d.update({ a: 2 }, { a: 3 }, {}, function () {
               d.find({}, function (err, docs) {
                 docs[0].a.should.equal(3);
-                done();
 
+                process.removeAllListeners('uncaughtException');
+                for (var i = 0; i < currentUncaughtExceptionHandlers.length; i += 1) {
+                  process.on('uncaughtException', currentUncaughtExceptionHandlers[i]);
+                }
```
Comment (JamesMGreene) [inline test/executor.test.js]: Keeping the listeners tidy.

File: test/executor.test.js [hunk 14]
```diff
@@ -51,29 +52,34 @@ function testRightOrder (d, done) {
...
+
+                done();
               });
             });
           });
-          
-          throw 'Some error';
+
+          throw new Error('Some error');
```
Comment (JamesMGreene) [inline test/executor.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/indexes.test.js [hunk 15]
```diff
@@ -120,107 +120,108 @@ describe('Indexes', function () {
       assert.deepEqual(idx.tree.search('world'), []);
       assert.deepEqual(idx.tree.search('bloup'), []);
     });
-
```
Comment (JamesMGreene) [inline test/indexes.test.js]: Correcting the indentation of this block.

File: test/persistence.test.js [hunk 16]
```diff
@@ -670,7 +670,7 @@ describe('Persistence', function () {
             fs.existsSync(testDb).should.equal(true);
             fs.existsSync(testDb + '~').should.equal(false);
             if (!contents.match(/^{"hello":"world","_id":"[0-9a-zA-Z]{16}"}\n$/)) {
-              throw "Datafile contents not as expected";
+              throw new Error("Datafile contents not as expected");
```
Comment (JamesMGreene) [inline test/persistence.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/persistence.test.js [hunk 17]
```diff
@@ -697,7 +697,7 @@ describe('Persistence', function () {
...
```
Comment (JamesMGreene) [inline test/persistence.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/persistence.test.js [hunk 18]
```diff
@@ -721,7 +721,7 @@ describe('Persistence', function () {
...
```
Comment (JamesMGreene) [inline test/persistence.test.js]: Using proper `Error` instances, a la previous PR #341.

File: test/persistence.test.js [hunk 19]
```diff
@@ -743,7 +743,7 @@ describe('Persistence', function () {
         fs.existsSync(dbFile).should.equal(true);
         fs.existsSync(dbFile + '~').should.equal(false);
         if (contents != "") {
-          throw "Datafile contents not as expected";
+          throw new Error("Datafile contents not as expected");
```
Comment (JamesMGreene) [inline test/persistence.test.js]: Using proper `Error` instances, a la previous PR #341.

**Comment by JamesMGreene** [comment]
@lukasbestle: Ideally true but I am just trying to make things obvious so the changes are more likely to get merged without any need for back-and-forth discussion. :wink:

File: test/db.test.js (hunk 10 above)
Comment (louischatriot) [inline test/db.test.js]: Indeed this is necessary

File: test/executor.test.js (hunk 11 above)
Comment (louischatriot) [inline test/executor.test.js]: Indeed, same as above.

File: test/indexes.test.js (hunk 15 above)
Comment (louischatriot) [inline test/indexes.test.js]: I was coding on Windows at that time (if you can believe it), and Windows gvim is not whitespace-friendly :)

**Comment by louischatriot** [comment]
Thanks for the PR, always good to have a second eye to look for this kind of stuff. Just merged it (except for two places where I reverted to my coding style), will be part of v1.6

**Comment by JamesMGreene** [comment]
You're welcomed, @louischatriot. Glad to help out! :+1:

**Pull request merged**